web/src/engine/websites/Alphapolis.ts

```diff
@@ -38,24 +32,34 @@ export default class extends DecoratableMangaScraper {
         return icon;
     }
 
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const results: Manga[] = [];
+        for (const character of '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ'.split('')) {
+            const mangas = await Common.FetchMangasMultiPageCSS.call(this, provider, `/search?category=official_manga&query=${character}&page={page}`, 'div.mangas-list div.wrap div.title a', 1, 1, 500);
+            results.push(...mangas);
+        }
+        results.push(...await Common.FetchMangasMultiPageCSS.call(this, provider, '/manga/index?sort=title&limit=1000&page={page}', 'div.content-main div.content-title a', 1, 1, 500));
+        return results.distinct();
+    }
+
     public override async FetchPages(chapter: Chapter): Promise<Page[]> {
-        const request = new Request(new URL(chapter.Identifier, this.URI));
-        const data = await FetchCSS(request, 'viewer-manga-horizontal');
+        const [viewer]= await FetchCSS(new Request(new URL(chapter.Identifier, this.URI)), '[v-bind\\:pages]');
         try {
-            const pages = JSON.parse(data[0].getAttribute('v-bind:pages'));
-            return pages.filter(element => typeof element != 'object' && !element.match('white_page')).map(element => new Page(this, chapter, new URL(element.replace(/\/[0-9]+x[0-9]+.([\w]+)/, '/1080x1536.$1')), {fallbackURL: element}));
+            const pages = JSON.parse(viewer.getAttribute('v-bind:pages'));
+            const isVertical = viewer.getAttribute('v-bind:is-vertical-manga') === '1';
+            return pages.filter(element => typeof element === 'string' && !element.match('white_page') && element != '').map(element => new Page(this, chapter, isVertical ? new URL( element ) : new URL(element.replace(/\/[0-9]+x[0-9]+(([./])[\w]+)/, '/1080x1536$1')), { fallbackURL: element }));
         } catch (error) {
             throw new Exception(R.Plugin_Common_Chapter_UnavailableError);
         }
     }
 
     //Since high resolution is not always available, use the real picture url instead of the forces one in case of failure
     public override async FetchImage(page: Page, priority: Priority, signal: AbortSignal): Promise<Blob> {
```

Comment:
I see.
Promise .any may return the lower quality image even if  the high quality promise succeded, :/

Reply:
Since a `try ... catch` is used to determine when the fallback link shall be used, i was assuming that there is no such case that both links can be valid.

Reply:
I wouldnt call a fallback url a real fallback url if it was invalid in some cases. Its the one given by the api after all :)